### PR TITLE
rgw: move keystone secrets from ceph.conf to files

### PR DESCRIPTION
**Pull request by mattbenjamin**
Permits setting restrictive permissions on these secrets.

Fixes: http://tracker.ceph.com/issues/36621

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@cbodley read_secret tested w/input of 9 and 2K

**Comment by cbodley** [comment]
code looks good. i like the deprecation strategy. could you please update http://docs.ceph.com/docs/master/radosgw/keystone/ to reflect the changes?

**Comment by mattbenjamin** [comment]
@cbodley added doc

File: doc/radosgw/config-ref.rst [hunk 1]
```diff
@@ -763,11 +763,26 @@ Keystone Settings
               authentication with the admin credentials
               (``rgw keystone admin user``, ``rgw keystone admin password``,
               ``rgw keystone admin tenant``, ``rgw keystone admin project``,
-              ``rgw keystone admin domain``). Admin token feature is considered
-              as deprecated.
+              ``rgw keystone admin domain``). The Keystone admin token
+              has been deprecated, but can be used to integrate with
+              older environments.Prefer ``rgw keystone admin token path``
```
Comment (cbodley) [inline doc/radosgw/config-ref.rst]: space after period

Comment (mattbenjamin) [reply doc/radosgw/config-ref.rst]: oops

**Comment by cbodley** [comment]
http://pulpito.ceph.com/cbodley-2018-11-07_17:48:22-rgw-wip-cbodley-testing-distro-basic-smithi/

**Pull request merged**